android-app/app/src/main/java/de/dhbw/ka/se/fibo/SplashActivity.java

```diff
@@ -25,6 +34,35 @@ protected void onCreate(Bundle savedInstanceState) {
         Objects.requireNonNull(supportActionBar);
         supportActionBar.hide();
 
+        if (ActivityUtils.isEspressoTesting()) {
+            SplashActivity.getIdlingResource().registerIdleTransitionCallback(this::startSync);
+        } else {
```

Comment:
Side note: The tests are no longer flaky. (As far as I can see.)

I think in other apps you don't see that data is loaded in the background is because you don't see the source code. ;) Out of my experience, it is actually quite common to load data as early as possible to avoid users to stare at a loading indicator when the work could be done before users see a loading indicator.

I don't want to debate about this, so for the sake of simplicity and also to have less sources of potential errors, let's get rid of this anyway. This would require that the MainActivity also receives the `SynchronizationResult`s even when it isn't the source of the request.

I'm going to remove this.